benchmark/profile_serving.py

```diff
@@ -1,198 +1,257 @@
-    warmup(tritonserver_addr, concurrency, session_len - 1)
-    req_que = mp.Queue()
-    res_que = mp.Queue()
-
-    procs = []
-    for i in range(concurrency):
-        chatbot = Chatbot(tritonserver_addr=tritonserver_addr,
-                          display=False,
-                          profile_serving=True,
-                          ignore_eos=True,
-                          log_level=logging.ERROR)
-        proc = mp.Process(target=infer,
-                          args=(chatbot, i + 1, req_que, res_que))
-        procs.append(proc)
-
-    # read data and put it to queue
-    n_req = read_dataset(tokenizer_path, dataset_path, samples, session_len,
-                         req_que)
-    for i in range(concurrency):
-        req_que.put([None, None, None])
-    _start = time.perf_counter()
-    for proc in procs:
-        proc.start()
-
-    stats = []
-    for i in range(concurrency):
-        session_id, _stats = res_que.get()
-        print(f'\n{"-" * 50}\n'
-              f'session {session_id}: processed reqs {len(_stats)}, '
-              f'stats: \n{_stats}\n{"-" * 50}\n')
-        stats.append(np.array(_stats))
-    _end = time.perf_counter()
-
-    elapsed_time = _end - _start
-
-    stats = np.concatenate(stats).reshape(-1, 5)
-
-    first_token_latency_min = np.min(stats[:, 0], axis=0)
-    first_token_latency_max = np.max(stats[:, 0], axis=0)
-    first_token_latency_ave = np.mean(stats[:, 0], axis=0)
-    completion_tokens = np.sum(stats[:, 1], axis=0)
-    request_output_tokens = np.sum(stats[:, 2], axis=0)
-    total_tokens = np.sum(stats[:, 3], axis=0)
-    prompt_tokens = total_tokens - completion_tokens
-    completion_token_throughput = completion_tokens / elapsed_time
-    total_token_throughput = total_tokens / elapsed_time
-    rqs = n_req / elapsed_time
-    rqm = rqs * 60
-
-    if (np.abs(stats[:, 1] - stats[:, 2]) <= 1).min() is False:
-        print(f'Did not generate requested number of tokens. '
-              f'Request {request_output_tokens:.0f}, '
-              f'but got {completion_tokens:.0f}')
-
-    print(
-        f'\n{"-" * 50}\nconcurrency: {concurrency}\n'
-        f'elapsed_time: {elapsed_time:.3f}s\n'
-        f'first_token latency(min, max, ave): '
-        f'{first_token_latency_min:.3f}s, {first_token_latency_max:.3f}s, '
-        f'{first_token_latency_ave:.3f}s\n'
-        f'number of prompt tokens: {prompt_tokens:.0f}\n'
-        f'number of completion tokens: {completion_tokens:.0f}\n'
-        f'token throughput (completion token): {completion_token_throughput:.3f} token/s\n'  # noqa
-        f'token throughput (prompt + completion token): {total_token_throughput:.3f} token/s\n'  # noqa
-        f'RPS (request per second): {rqs:.3f} req/s\n'
-        f'RPM (request per minute): {rqm:.3f} req/min\n'
-        f'{"-" * 50}\n')
-    for proc in procs:
-        proc.join()
+         dataset: str,
+         concurrency: int = 32,
+         num_prompts: int = 1000,
```

Comment:
make it the same as the value of the other two scripts?

Reply:
It cannot be since the `instance_group`  in `interactive/config.pbtxt` defaults to 48